lib/mountable_file_server/metadata.rb

```diff
@@ -0,0 +1,45 @@
+require 'dimensions'
+
+module MountableFileServer
+  class Metadata
+    attr_reader :size, :content_type, :height, :width
+
+    def initialize(size:, content_type:, width: nil, height: nil)
+      @size = size
+      @content_type = content_type
+      @width = width
+      @height = height
+    end
+
+    def to_h
+      hash = {
+        size: size,
+        content_type: content_type
+      }
+
+      if width && height
+        hash = hash.merge({
+          height: height,
+          width: width
+        })
+      end
+
+      hash
+    end
+
+    def self.for_path(path)
+      parameters = {}
+
+      parameters[:content_type] = `file --brief --mime-type #{path}`.strip
+      parameters[:size] = File.size(path)
+
+      if ['image/png', 'image/jpeg', 'image/gif'].include?(parameters[:content_type])
```

Comment:
Those MIME types are OK for now and will presumably suffice for most of our use cases but maybe it´s worth a thought to make this configurable later (with a default value).